packages/liveblocks-core/src/connection.ts

```diff
@@ -506,20 +509,14 @@ function createConnectionStateMachine<T extends BaseAuthResult>(
   };
 
   machine
-    .onEnter("@ok.*", (ctx) => {
+    .onEnter("@ok.*", () => {
       // Do nothing on entering...
 
       // ...but when *leaving* OK state, always tear down the old socket. It's
       // no longer valid.
-      return () => {
+      return (ctx) => {
         teardownSocket(ctx.socket);
-        (ctx as any).socket = null;
-        //   ^^^^^^
-        //   XXX Really, I should add an official API to the FSM to support
-        //   XXX assigning to context when entering/leaving a state, since this
-        //   XXX is a well-defined moment in a state transition.
-        //   XXX I've only temporarily pulled this hack here, for which I ask
-        //   XXX forgiveness.
+        ctx.patch({ socket: null });
```

Comment:
Here, in the new setup, if we'd accidentally use `ctx` from the onEnter callback, that would throw an error and catch that bug for you.